.circleci/config.yml

```diff
@@ -0,0 +1,54 @@
+version: 2.1
+
+executors:
+  my-docker-executor:
+    docker:
+      - image: cimg/python:3.11.10
+
+jobs:
+  build:
+    executor: my-docker-executor
+    steps:
+      - checkout
+
+      - run:
+          name: Get uv via curl and install it
+          command: |
+            curl -LsSf https://astral.sh/uv/install.sh | sh
+
+      - persist_to_workspace:
+          root: /home/circleci/
+          paths:
+            - .cargo
+
+  test:
+    executor: my-docker-executor
+    steps:
+      - checkout
+
+      - attach_workspace:
+          at: /home/circleci/
+
+      - run:
+          name: Set-up dependencies path
+          command: echo 'export PATH=$HOME/.cargo/bin:$PATH' >> $BASH_ENV
+
+      - run:
+          name: Set-up project dependencies
+          command: uv sync
+
+      - run:
+          name: Format python files with ruff
+          command: uv run ruff check . --fix
```

Comment:
I found only four check commands for ruff.
`ruff check                  # Lint all files in the current directory.`
`ruff check --fix            # Lint all files in the current directory, and fix any fixable errors.`
`ruff check --watch          # Lint all files in the current directory, and re-lint on change.`
`ruff check path/to/code/    # Lint all files in `path/to/code` (and any subdirectories).`

Even though the --fix fixes safe fixes, it might broke code because of any bugs in the ruff package. I would fix it in the next version.